ts/deploy.ts

```diff
@@ -170,6 +165,28 @@ export async function deployAll(
         nameRegistry,
         await paramManager.TEST_TOKEN()
     );
+    await tokenRegistry.requestTokenRegistration(testToken.address);
+    await tokenRegistry.finaliseTokenRegistration(testToken.address);
+
+    const spokeRegistry = await new SpokeRegistryFactory(signer).deploy();
+    await waitAndRegister(
+        spokeRegistry,
+        "spokeRegistry",
+        verbose,
+        nameRegistry,
+        await paramManager.SPOKE_REGISTRY()
+    );
+
+    const vault = await new VaultFactory(allLinkRefs, signer).deploy(
```

Comment:
Gotchas here: 
Vault depends on spokeRegistry and Deposit Manager depends on Vault. If we deploy with the wrong order, the transactions fail without giving any reason.
 Created an issue in #270
A workaround is always requiring non-Zero contract address in the constructor.